applicationset/generators/scm_provider.go

```diff
@@ -230,6 +230,15 @@ func (g *SCMProviderGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha
 			"branchNormalized": utils.SanitizeName(repo.Branch),
 		}
 
+		if providerConfig.Gitlab != nil {
+			latestProtectedTag, err := scm_provider.GitlabLatestProtectedTag(ctx, repo, provider.(*scm_provider.GitlabProvider))
```

Comment:
If it's possible to do this without an extra API call, I'd try to go that route. If not, I wonder if this should be opt-in to avoid negative performance impacts on users who don't need the tags.

Reply:
It's not possible to do without an extra API call because you need to obtain additional information about tags in GitLab.
But you are right it should opt-in. I will implement it.